frontend/src/components/forms/CreateEmployee.tsx

```diff
@@ -0,0 +1,320 @@
+
+  const handleFirstNameChange = (e: { target: { value: unknown } }) => {
+    const inputValue: string = e.target.value as string;
+    setInvitedFirstName(inputValue);
+    if (inputValue) {
+      setInvitedFirstNameError(false);
+    } else {
+      setInvitedFirstNameError(true);
+    }
+  };
+
+  const handleLastNameChange = (e: { target: { value: unknown } }) => {
+    const inputValue: string = e.target.value as string;
+    setInvitedLastName(inputValue);
+    if (inputValue) {
+      setInvitedLastNameError(false);
+    } else {
+      setInvitedLastNameError(true);
+    }
+  };
+
+  const handleEmailChange = (e: { target: { value: unknown } }) => {
+    const inputValue: string = e.target.value as string;
+    setInvitedEmail(inputValue);
+    if (inputValue.endsWith("@supportivehousing.com")) {
+      setInvitedEmailError(false);
+    } else {
+      setInvitedEmailError(true);
+    }
+  };
+
+  const handleAdminStatusChange = (inputValue: string) => {
+    setinvitedAdminStatus(inputValue);
+    if (inputValue !== "") {
+      setInvitedAdminStatusError(false);
+    } else {
+      setInvitedAdminStatusError(true);
+    }
+  };
+
+  const handleOpen = () => {
+    setIsOpen(true);
+    setInvitedEmail("");
+    setInvitedFirstName("");
+    setInvitedLastName("");
+    setinvitedAdminStatus("");
+    setIsTwoFactorAuthenticated(false);
+
+    setInvitedEmailError(false);
+    setInvitedFirstNameError(false);
+    setInvitedLastNameError(false);
+    setInvitedAdminStatusError(false);
+  };
+
+  const handleClose = () => {
+    setIsOpen(false);
+  };
+
+  const onInviteEmployee = async (
+    isEmailError: boolean,
+    isFirstNameError: boolean,
+    isLastNameError: boolean,
+    isAdminStatusError: boolean,
+  ) => {
+    if (
+      !isEmailError &&
+      !isFirstNameError &&
+      !isLastNameError &&
+      !isAdminStatusError
+    ) {
+      let hasInvitedUser = false;
+      if (invitedAdminStatus === "1") {
+        hasInvitedUser = await commonApiClient.inviteUser(
+          invitedEmail,
+          RoleOptions[1],
+          invitedFirstName,
+          invitedLastName,
+        );
+      } else if (invitedAdminStatus === "2" && isTwoFactorAuthenticated) {
+        hasInvitedUser = await commonApiClient.inviteUser(
+          invitedEmail,
+          RoleOptions[0],
+          invitedFirstName,
+          invitedLastName,
+        );
+      } else if (invitedAdminStatus === "2" && !isTwoFactorAuthenticated) {
+        hasInvitedUser = await commonApiClient.inviteUser(
+          invitedEmail,
+          RoleOptions[2],
+          invitedFirstName,
+          invitedLastName,
+        );
+      }
+      if (hasInvitedUser) {
+        newToast(
+          "Invite sent",
+          `Your invite has been sent to ${invitedFirstName} ${invitedLastName}`,
+          "success",
+        );
+        handleClose();
+      }
+    } else {
+      newToast("Error inviting employee", INVITE_EMPLOYEE_ERROR, "error");
```

Comment:
we don't need to show a toast message here because the form validation is enough to indicate to the user that something is invalid BUT its good practice to add error messages when needed :')

Some better use cases of displaying an error toast message is if the form is valid but somehow the submission failed (api returns an error)

Reply:
I caught an error where this toast message could be used, it's when you invite a user who's already in the system. The error should be caught and handled in the back-end and displayed here, but the check was written in the service layer of the API so we should do that :') we can sync on how to fix this